lib/flux-core/src/iter/traits/iterator.rs

```diff
@@ -0,0 +1,33 @@
+use flux_attrs::*;
+
+#[extern_spec(core::iter)]
+#[assoc(fn valid_item(self: Self, item: Self::Item) -> bool { true })]
+#[assoc(fn size(self: Self) -> int)]
+#[assoc(fn done(self: Self) -> bool)]
+#[assoc(fn step(self: Self, other: Self) -> bool { true })]
```

Comment:
You can put these in a single block btw. I find it a bit more readable. 

Also, we are using `assoc` in some places and `reft` in others; they both mean the same, but we should probably pick one. I don't have strong opinions

```suggestion
#[assoc(
    fn valid_item(self: Self, item: Self::Item) -> bool { true }
    fn size(self: Self) -> int);
    fn done(self: Self) -> bool);
    fn step(self: Self, other: Self) -> bool { true }
)]
```